### PR TITLE
Add wind tests to the test suite (non dusty winds)

### PR DESCRIPTION
**Pull request by lsiess**
Type of PR: 
new test_wind.f90 routine

Description:
Provide a test suite for non dusty  winds from a single star

Testing:
I ran make test and passed all the tests

Did you run the bots? yes

Did you update relevant documentation in the docs directory? no

### CODE REVIEW

File: src/tests/test_wind.f90 [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+ else
+    gamma = 1.66666667
+    ieos = 2
+ endif
+ !polyk = kboltz*T_wind/(mass_proton_cgs * gmw * unit_velocity**2)
+
+ call init_eos(ieos,ierr)
+
+ iverbose = 1
+ hfact    = 1.2
```
Comment (danieljprice) [inline src/tests/test_wind.f90:94]: hfact should be hfact = hfact_default from the kernel module here

File: src/tests/test_wind.f90 [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+ tolv     = 1.e-2
```
Comment (danieljprice) [inline src/tests/test_wind.f90:95]: suggest to not explicitly override input file variables which are the same as the default values, i.e. we should fail the tests if the default values set elsewhere in the code cause this test to fail

File: src/tests/test_wind.f90 [hunk 3]
```diff
@@ -0,0 +1,179 @@
+!--------------------------------------------------------------------------!
+! The Phantom Smoothed Particle Hydrodynamics code, by Daniel Price et al. !
+! Copyright (c) 2007-2023 The Authors (see AUTHORS)                        !
+! See LICENCE file for usage and distribution conditions                   !
+! http://phantomsph.bitbucket.io/                                          !
+!--------------------------------------------------------------------------!
+module testwind
+!
+! Unit tests of the step module
```
Comment (danieljprice) [inline src/tests/test_wind.f90:9]: wind injection module

**Comment by danieljprice** [comment]
some of the failing tests in #450 will also be fixed by merging this...

**Comment by danieljprice** [comment]
build failures are in phantomtest, e.g.:
```
 Checking balsarakim (phantomtest)... FAILED
  Error: More actual than formal arguments in procedure call at (1)
  make[1]: *** [Makefile:462: test_wind.o] Error 1
  make: *** [Makefile:17: phantomtest] Error 2
```
i.e. can reproduce with:
```
make phantomtest SETUP=balsarakim
```

**Comment by lsiess** [comment]
I really don't understand what's wrong ? The only thing I changed with respect to the previous commit was to add call init_step to fixed the remaining problem with the SETUP=testkd and now most of the MPI tests fail ????
I am not familiar with MPI. Could you have a look. I am out of ideas. 
Sorry and thanks!

**Comment by lsiess** [comment]
I really don't understand what's wrong ? The only thing I changed with respect to the previous commit was to add `call init_step` to fixed the remaining problem with SETUP=testkd and now most of the MPI tests fail ????
I am not familiar with MPI. Could you have a look. I am out of ideas.

Also, on my laptop when I compile
` make SETUP=test MPI=yes DEBUG=no SYSTEM=gfortran OPENMP=no phantomtest`
I have many error messages 
```
mpif90 `mpif90 --showme:compile` -c -O3 -Wall -Wno-unused-dummy-argument -frecord-marker=4 -gdwarf-2 -finline-functions-called-once -finline-limit=1500 -funroll-loops -ftree-vectorize -std=f2008 -fall-intrinsics -fPIC -fdefault-real-8 -fdefault-double-8 ../src/tests/phantomtest.f90 -o phantomtest.o
gfortran: error: unrecognized command-line option ‘--showme:compile’
```
I am using
 `GNU Fortran (Ubuntu 12.2.0-17ubuntu1) 12.2.0`

sorry and thanks for your help

**Comment by danieljprice** [comment]
I think it's just that the new wind test is not working in the MPI code, it's the wind test that's failing.

In other words this is a real problem, namely that the wind injection is currently not implemented properly in MPI. I can take a look at this.

**Comment by danieljprice** [comment]
the "build" failures are just some unused variable warnings left in test_wind.f90, since we now disallow compiler warnings on the master branch... you can reproduce these with 
```
make NOWARN=yes
```
giving

```
Error: Unused module variable ‘icooling’ which has been explicitly imported at (1) [-Werror=unused-variable]
  Error: Unused module variable ‘iverbose’ which has been explicitly imported at (1) [-Werror=unused-variable]
  make[1]: *** [Makefile:462: test_wind.o] Error 1
 ```

**Comment by danieljprice** [comment]
am going to merge this into #451 so they become one big pull request, then I can fix things from my end

**Pull request merged**